File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/Procedure.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Procedure : DomainResource, IResource
{
    public Base.CodeableConcept? Category { get; set; }
    public Base.ResourceReference[]? Report { get; set; }
    public Base.CodeableConcept[]? UsedCode { get; set; }
    public Base.ResourceReference[]? UsedReference { get; set; }
    public string[]? InstantiatesCanonical { get; set; }
    public string[]? InstantiatesUri { get; set; }
    public Procedure_FocalDevice[]? FocalDevice { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public string? PerformedAge { get; set; }
    public Base.ResourceReference[]? ComplicationDetail { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public string? PerformedString { get; set; }
    public Base.CodeableConcept? StatusReason { get; set; }
    public object? Performed    
    {
        get
        {
            if (PerformedAge is not null)
            {
                return PerformedAge;
            }
    
            if (PerformedString is not null)
            {
                return PerformedString;
            }
    
            if (PerformedRange is not null)
            {
                return PerformedRange;
            }
    
            if (PerformedDateTime is not null)
            {
                return PerformedDateTime;
            }
    
            if (PerformedPeriod is not null)
            {
                return PerformedPeriod;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                PerformedAge = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                PerformedString = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.Range))
            {
                PerformedRange = (Base.Range)value;
                return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                PerformedDateTime = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                PerformedPeriod = (Base.Period)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.CodeableConcept? Outcome { get; set; }
    public Base.ResourceReference? Asserter { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.Range? PerformedRange { get; set; }
    public Base.CodeableConcept[]? Complication { get; set; }
    public required string Status { get; set; }
    public string? PerformedDateTime { get; set; }
    public Base.ResourceReference? Recorder { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.CodeableConcept[]? BodySite { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.Period? PerformedPeriod { get; set; }
    public Base.ResourceReference? Location { get; set; }
    public Base.CodeableConcept[]? FollowUp { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public Procedure_Performer[]? Performer { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class Procedure_FocalDevice : BackboneElement
    {
        public Base.CodeableConcept? Action { get; set; }
        public required Base.ResourceReference Manipulated { get; set; }
    }

    public class Procedure_Performer : BackboneElement
    {
        public required Base.ResourceReference Actor { get; set; }
        public Base.CodeableConcept? Function { get; set; }
        public Base.ResourceReference? OnBehalfOf { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Procedure.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Age
from base import Period
from base import CodeableConcept
from base import Range
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ProcedureFocalDevice(BackboneElement):
    action: Optional[CodeableConcept] = None
    manipulated: Reference

@dataclass(kw_only=True)
class ProcedurePerformer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None
    on_behalf_of: Optional[Reference] = None

@dataclass(kw_only=True)
class Procedure(DomainResource):
    category: Optional[CodeableConcept] = None
    report: Optional[List[Reference]] = field(default_factory=list)
    used_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    used_reference: Optional[List[Reference]] = field(default_factory=list)
    instantiates_canonical: Optional[List[str]] = field(default_factory=list)
    instantiates_uri: Optional[List[str]] = field(default_factory=list)
    focal_device: Optional[List[ProcedureFocalDevice]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    performed_age: Optional[Age] = None
    complication_detail: Optional[List[Reference]] = field(default_factory=list)
    reason_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    performed_string: Optional[str] = None
    status_reason: Optional[CodeableConcept] = None
    outcome: Optional[CodeableConcept] = None
    asserter: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    performed_range: Optional[Range] = None
    complication: Optional[List[CodeableConcept]] = field(default_factory=list)
    status: str
    performed_date_time: Optional[str] = None
    recorder: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    body_site: Optional[List[CodeableConcept]] = field(default_factory=list)
    based_on: Optional[List[Reference]] = field(default_factory=list)
    part_of: Optional[List[Reference]] = field(default_factory=list)
    performed_period: Optional[Period] = None
    location: Optional[Reference] = None
    follow_up: Optional[List[CodeableConcept]] = field(default_factory=list)
    subject: Reference
    performer: Optional[List[ProcedurePerformer]] = field(default_factory=list)
    reason_reference: Optional[List[Reference]] = field(default_factory=list)